zcash_keys/src/keys.rs

```diff
@@ -1105,6 +1161,50 @@ impl UnifiedIncomingViewingKey {
         }
     }
 
+    fn from_checked_parts(
```

Comment:
`UnifiedIncomingViewingKey::new` should probably be altered to use this method, as currently (when `test-dependencies` is enabled`) it enables producing invalid states.